apps/tlon-web/e2e/gallery-functionality.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { expect, test } from '@playwright/test';
+
+import * as helpers from './helpers';
+import shipManifest from './shipManifest.json';
+
+const zodUrl = `${shipManifest['~zod'].webUrl}/apps/groups/`;
+
+test.use({ storageState: shipManifest['~zod'].authFile });
+
+test('should test gallery functionality', async ({ page }) => {
+  await page.goto(zodUrl);
+  await helpers.clickThroughWelcome(page);
+  await page.evaluate(() => {
+    window.toggleDevTools();
+  });
+
+  // Assert that we're on the Home page
+  await expect(page.getByText('Home')).toBeVisible();
+
+  // Clean up any existing group on zod
+  await helpers.cleanupExistingGroup(page, 'Test Group');
+  await helpers.cleanupExistingGroup(page, '~ten, ~zod');
+  await helpers.cleanupExistingGroup(page);
+
+  // Create a new group
+  await helpers.createGroup(page);
+
+  // Navigate back to Home and verify group creation
+  await helpers.navigateBack(page);
+
+  if (await page.getByText('Home').isVisible()) {
+    await expect(page.getByText('Untitled group')).toBeVisible();
+    await page.getByText('Untitled group').click();
+    await expect(page.getByText('Untitled group').first()).toBeVisible();
+  }
+
+  // Create a new gallery channel
+  await helpers.openGroupSettings(page);
+  await page.getByTestId('GroupChannels').getByText('Channels').click();
+  await helpers.createChannel(page, 'Test Gallery', 'gallery');
+
+  // Navigate to the gallery channel
+  await page.getByTestId('ChannelListItem-Test Gallery').click();
+
+  // Create a new gallery post
+  await helpers.createGalleryPost(page, 'Test Gallery Post Content');
```

Comment:
out of scope, we'll probably also want to test adding different kinds of gallery blocks

Reply:
Yeah, I was thinking about how we might want to go about testing image uploads here (and elsewhere). Maybe we could setup a minio server somewhere and add point the fakeships at that.

Links would be easy to add, though.